src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java

```diff
@@ -351,5 +352,76 @@ static UMessageType getMessageType(String ce_type){
                 return UMessageType.UMESSAGE_TYPE_UNSPECIFIED;
         }
     }
+    /**
+     * Get the UMessage from the cloud event
+     * @param event The CloudEvent containing the data.
+     * @return returns the UMessage
+     */
+    static UMessage toMessage(CloudEvent event) {
+        UUri source = LongUriSerializer.instance().deserialize(getSource(event));
+        UPayload payload = UPayload.newBuilder().setFormat(UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF)
+                .setValue(getPayload(event).toByteString()).build();
+        UAttributes.Builder builder =
+                UAttributes.newBuilder().setId(LongUuidSerializer.instance().deserialize(event.getId()))
+                        .setType(getMessageType(event.getType())).setTtl(getCommunicationStatus(event))
+                        .setPriority(UPriority.valueOf(getPriority(event).toString()));
+        if (getSink(event).isPresent()) {
+            builder.setSink(LongUriSerializer.instance().deserialize(getSink(event).get()));
+        }
+        if (getTtl(event).isPresent()) {
+            builder.setTtl(getTtl(event).get());
+        }
+        if (getRequestId(event).isPresent()) {
+            builder.setReqid(LongUuidSerializer.instance().deserialize(getRequestId(event).get()));
+        }
+        if (getToken(event).isPresent()) {
+            builder.setToken(getToken(event).get());
+        }
+        Integer permission_level = extractIntegerValueFromExtension("permission_level", event).orElse(UCode.OK_VALUE);
+        builder.setPermissionLevel(permission_level);
+        UAttributes attributes=builder.build();
+
+        return UMessage.newBuilder().setAttributes(attributes).setPayload(payload).setSource(source).build();
+
+    }
+
+    /**
+     * Get the Cloudevent from the UMessage
+     * @param message The UMessage protobuf containing the data
+     * @return returns the cloud event
+     */
+    static CloudEvent fromMessage(UMessage message) {
+        UAttributes attributes = message.getAttributes();
```

Comment:
I wonder if you should use the CloudEventFactory or is the issue circular dependencies?

Reply:
No circular dependency was found, but I decided to remove it anyway